libflux/flux-core/src/semantic/types.rs

```diff
@@ -882,6 +882,8 @@ impl MonoType {
             // An error has already occurred so assume everything is ok here so that we do not
             // create additional, spurious errors
             (MonoType::Error, _) | (_, MonoType::Error) => (),
+            // Similarly if both sides are dynamic, there's nothing more to do.
+            (MonoType::Dynamic(_), MonoType::Dynamic(_)) => (),
```

Comment:
#5239 landed moments ago, so you should be able to drop that cherry-picked commit :crossed_fingers: